### PR TITLE
Update cereal to v1.2.1 w/ boost Archive::is_{loading,saving} compatibility

### PR DESCRIPTION
**Pull request by headupinclouds**

### CODE REVIEW

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1194
* https://travis-ci.org/ingenue/hunter/builds/179052500

File: cmake/projects/cereal/hunter.cmake [hunk 1]
```diff
@@ -12,6 +12,17 @@ include(hunter_cacheable)
...
 
+hunter_add_version(
+    PACKAGE_NAME
+    cereal
+    VERSION
+    "1.2.1-p1"
+    URL
+    "https://github.com/headupinclouds/cereal/archive/v1.2.1.tar.gz"    
+    SHA1
+    8d8179d71a12bafae3d90b6a5a7fc09cb2e69474
```
Comment (ruslo) [inline cmake/projects/cereal/hunter.cmake]: Should be `3b1188758515ed645e9ca073f7077828bdffa41d`

Comment (headupinclouds) [inline cmake/projects/cereal/hunter.cmake]: Sorry about that.  Fixed.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1195
* https://travis-ci.org/ingenue/hunter/builds/179059292

**Comment by ruslo** [comment]
https://github.com/headupinclouds/cereal/commit/433547db44e9a1c08372a2432d8f8c281c297f76

**Comment by headupinclouds** [comment]
I've updated to the v1.2.1-p1 release w/ matching SHA1.  Sorry for the confusion.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1196
* https://travis-ci.org/ingenue/hunter/builds/179095172

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1197
* https://travis-ci.org/ingenue/hunter/builds/179100490

**Comment by ruslo** [comment]
> Uploading

Done. Released: https://github.com/ruslo/hunter/releases/tag/v0.16.33

**Comment by headupinclouds** [comment]
Thanks.

**Pull request merged**